textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -34,12 +35,23 @@
     private IOperator inputOperator;
     private String query;
 
-    public KeywordMatcher(IPredicate predicate) {
-        this.predicate = (KeywordPredicate)predicate;
-        DataReaderPredicate dataReaderPredicate = this.predicate.getDataReaderPredicate();
+    public KeywordMatcher(IPredicate predicate, IDataStore dataStore) {
```

Comment:
Per our discussion, let's push the logic of using `dataStore` into the constructor of `IndexBasedSourceOperator`.  A caller of `KeywordMatcher()` needs to construct an `IndexBasedSourceOperator` explicitly, then pass it to the constructor of  `KeywordMatcher`.